lib/everypolitician.rb

```diff
@@ -39,6 +43,18 @@ def self.country_legislature(country_slug, legislature_slug)
     [country, legislature]
   end
 
+  def self.from_sources_dir(dir)
+    @index_by_sources ||= begin
+      sources = {}
+      parsed_json.each do |c|
+        country = Country.new(c)
+        country.legislatures.each { |l| sources[l.directory] = [ country, l ] }
+      end
+      sources
+    end
+    @index_by_sources[dir]
+  end
+
```

Comment:
I think you could simplify both the interface and implementation greatly here.

As a `legislature` already has a `.country` method defined on it, it seems needless to return both a `country`, and `legislature`. This could all then be simplified greatly to just:

```ruby
@index_by_sources ||= countries.map(&:legislatures).flatten.group_by(&:directory)
@index_by_sources[dir]
```

(Even if you did need to return an array of both the country and the legislature, you could still simplify the construction of `@index_by_sources` similarly to this. In general it's a red flag in ruby to create an empty variable and then assign to it in a loop.)